src/python_inspector/resolution.py

```diff
@@ -134,21 +170,116 @@ def get_requirements_for_package(self, purl, candidate):
+                    repos=self.repos,
+                )
+                for wheel in wheels:
+                    deps = list(
+                        PypiWheelHandler.parse(os.path.join(utils_pypi.CACHE_THIRDPARTY_DIR, wheel))
+                    )
+                    assert len(deps) == 1
+                    deps = deps[0].dependencies
+                    for dep in deps:
+                        if dep.scope == "install":
+                            yield packaging.requirements.Requirement(str(dep.extracted_requirement))
+
+            if format == "Sdist":
+                sdist = utils_pypi.download_sdist(
+                    name=candidate.name,
+                    version=str(candidate.version),
+                    repos=self.repos,
+                )
+                if sdist.endswith(".tar.gz"):
+                    sdist_file = sdist.rstrip(".tar.gz")
+                    file = tarfile.open(os.path.join(utils_pypi.CACHE_THIRDPARTY_DIR, sdist))
+                    file.extractall(
+                        os.path.join(
+                            utils_pypi.CACHE_THIRDPARTY_DIR, "extracted_sdists", sdist_file
+                        )
+                    )
+                    file.close()
+                if sdist.endswith(".zip"):
+                    sdist_file = sdist.rstrip(".zip")
+                    with ZipFile(os.path.join(utils_pypi.CACHE_THIRDPARTY_DIR, sdist), "r") as zip:
+                        zip.extractall(
+                            os.path.join(
+                                utils_pypi.CACHE_THIRDPARTY_DIR, "extracted_sdists", sdist_file
+                            )
+                        )
+                setup_py_path = os.path.join(
+                    utils_pypi.CACHE_THIRDPARTY_DIR,
+                    "extracted_sdists",
+                    sdist_file,
+                    sdist_file,
+                    "setup.py",
+                )
+                setup_cfg_path = os.path.join(
+                    utils_pypi.CACHE_THIRDPARTY_DIR,
+                    "extracted_sdists",
+                    sdist_file,
+                    sdist_file,
+                    "setup.cfg",
+                )
+                pkg_info_path = os.path.join(
+                    utils_pypi.CACHE_THIRDPARTY_DIR, "extracted_sdists", sdist_file
+                )
+                requirement_path = os.path.join(
+                    utils_pypi.CACHE_THIRDPARTY_DIR,
+                    "extracted_sdists",
+                    sdist_file,
+                    sdist_file,
+                    "requirements.txt",
+                )
+
+                path_by_format = {
+                    "pkginfo": pkg_info_path,
+                    "setup-py": setup_py_path,
+                    "setup-cfg": setup_cfg_path,
+                    "requirement": requirement_path,
+                }
+
+                handler_by_format = {
+                    "pkginfo": PythonSdistPkgInfoFile,
```

Comment:
Each of these handlers has a "datasource_id" attribute... it would be a better key that a new made up one. For instance:
```suggestion
                    PythonSdistPkgInfoFile.datasource_id: PythonSdistPkgInfoFile,
                    .... and so on
```